pkg/migrations/sql/preinstall/008-tracing.sql

```diff
@@ -0,0 +1,253 @@
+GRANT USAGE ON DOMAIN SCHEMA_TRACING_PUBLIC.tag_type TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.span_tag_type()
+RETURNS SCHEMA_TRACING_PUBLIC.tag_type
+AS $sql$
+    SELECT (1<<0)::smallint::SCHEMA_TRACING_PUBLIC.tag_type
+$sql$
+LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.span_tag_type() TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.resource_tag_type()
+RETURNS SCHEMA_TRACING_PUBLIC.tag_type
+AS $sql$
+    SELECT (1<<1)::smallint::SCHEMA_TRACING_PUBLIC.tag_type
+$sql$
+LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.resource_tag_type() TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.event_tag_type()
+RETURNS SCHEMA_TRACING_PUBLIC.tag_type
+AS $sql$
+    SELECT (1<<2)::smallint::SCHEMA_TRACING_PUBLIC.tag_type
+$sql$
+LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.event_tag_type() TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.link_tag_type()
+RETURNS SCHEMA_TRACING_PUBLIC.tag_type
+AS $sql$
+    SELECT (1<<3)::smallint::SCHEMA_TRACING_PUBLIC.tag_type
+$sql$
+LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.link_tag_type() TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.is_span_tag_type(_tag_type SCHEMA_TRACING_PUBLIC.tag_type)
+RETURNS BOOLEAN
+AS $sql$
+    SELECT _tag_type & SCHEMA_TRACING_PUBLIC.span_tag_type() = SCHEMA_TRACING_PUBLIC.span_tag_type()
+$sql$
+LANGUAGE SQL IMMUTABLE STRICT PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.is_span_tag_type(SCHEMA_TRACING_PUBLIC.tag_type) TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.is_resource_tag_type(_tag_type SCHEMA_TRACING_PUBLIC.tag_type)
+RETURNS BOOLEAN
+AS $sql$
+    SELECT _tag_type & SCHEMA_TRACING_PUBLIC.resource_tag_type() = SCHEMA_TRACING_PUBLIC.resource_tag_type()
+$sql$
+LANGUAGE SQL IMMUTABLE STRICT PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.is_resource_tag_type(SCHEMA_TRACING_PUBLIC.tag_type) TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.is_event_tag_type(_tag_type SCHEMA_TRACING_PUBLIC.tag_type)
+RETURNS BOOLEAN
+AS $sql$
+    SELECT _tag_type & SCHEMA_TRACING_PUBLIC.event_tag_type() = SCHEMA_TRACING_PUBLIC.event_tag_type()
+$sql$
+LANGUAGE SQL IMMUTABLE STRICT PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.is_event_tag_type(SCHEMA_TRACING_PUBLIC.tag_type) TO prom_reader;
+
+CREATE OR REPLACE FUNCTION SCHEMA_TRACING_PUBLIC.is_link_tag_type(_tag_type SCHEMA_TRACING_PUBLIC.tag_type)
+RETURNS BOOLEAN
+AS $sql$
+    SELECT _tag_type & SCHEMA_TRACING_PUBLIC.link_tag_type() = SCHEMA_TRACING_PUBLIC.link_tag_type()
+$sql$
+LANGUAGE SQL IMMUTABLE STRICT PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION SCHEMA_TRACING_PUBLIC.is_link_tag_type(SCHEMA_TRACING_PUBLIC.tag_type) TO prom_reader;
```

Comment:
why are these functions in preinstall and not idempotent dirs.

Reply:
I needed the non-"is_" functions for the inserts of the standard tags, but i guess i could make the inserts on conflict do nothing and put it all in idempotent

Reply:
gonna leave inserts in preinstall and use magic numbers with a comment to explain them